presto-spi/src/main/java/io/prestosql/spi/predicate/SortedRangeSet.java

```diff
@@ -181,6 +182,9 @@ public boolean isDiscreteSet()
     @Override
     public boolean containsValue(Object value)
     {
+        if (isFloatingPointNaN(type, value)) {
```

Comment:
different commit?

Reply:
I will rename `Handle NaN when extracting predicate` to better describe its holistic approach